CMakeLists.txt

```diff
@@ -154,6 +154,13 @@ elseif("$ENV{ROS_VERSION}" STREQUAL "2")
       ros2_foxglove_bridge/src/parameter_interface.cpp
       ros2_foxglove_bridge/src/generic_client.cpp
     )
+
+    # Check if the ROS_DISTRO is greater than iron and set a compile definition if that's the case.
+    string(COMPARE GREATER $ENV{ROS_DISTRO} "iron" ROS_DISTRO_GT_IRON)
```

Comment:
https://cmake.org/cmake/help/latest/command/string.html#compare docs are so helpful on what this does /sarcasm :(

Reply:
I think technically yes because ROS distro names go in alphabetical order for ROS2 (but letters are re-used from ROS1) but I also think we might want something more robust than this.

Reply:
Also not super happy with this approach but not sure what else to do. I could check if ROS_DISTRO == rolling but then we will have to change this everytime a new release is spun off. Any other ideas?

Reply:
Can you use `target_compile_definitions(... PUBLIC RCLCPP_VERSION_MAJOR=${rclcpp_VERSION_MAJOR})`?

Reply:
That seems to work, great suggestion